lib/internal/modules/cjs/loader.js

```diff
@@ -480,7 +483,9 @@ Module._resolveLookupPaths = function(request, parent, newReturn) {
 
   // make sure require('./path') and require('path') get distinct ids, even
   // when called from the toplevel js file
-  if (parentIdPath === '.' && id.indexOf('/') === -1) {
+  if (parentIdPath === '.' &&
+      id.indexOf('/') === -1 &&
+      (!isWindows || id.indexOf('\\') === -1)) {
     id = './' + id;
   }
```

Comment:
This change looked necessary, but this piece of code does not seem to be used anywhere in node core nor tested. `newReturn` seems to be passed as true everywhere. Hence, I'm not sure about this, let me know if I should leave this as it was.

Reply:
`newReturn` is for backwards compatibility. There were some people using the private, underscored methods directly and were expecting the old return value.